### PR TITLE
Migrate from * to Type!

### PR DESCRIPTION
**Pull request by TheWizardTower**
I came across some lint nags as I was working on the persistent package, and thought I'd tidy them up.

### CODE REVIEW

File: core-data/lib/Core/Data/Structures.hs [hunk 1]
```diff
@@ -194,12 +195,12 @@ instance for @[(κ,ν)]@ lists shown above.
...
 --
 class Dictionary α where
-    type K α :: *
-    type V α :: *
+    type K α :: Type
+    type V α :: Type
     fromMap :: Map (K α) (V α) -> α
     intoMap :: α -> Map (K α) (V α)
 
-instance Key κ => Dictionary (Map κ ν) where
```
Comment (istathar) [inline core-data/lib/Core/Data/Structures.hs:202]: I'm a bit reluctant to remove this constraint, as it makes for a uniform view if the requirement in the instances here:
https://hackage.haskell.org/package/core-data-0.3.0.2/docs/Core-Data-Structures.html#t:Dictionary

**Review by istathar — COMMENTED** [review]
Maybe we should suppress the warning instead.

File: core-data/lib/Core/Data/Structures.hs (hunk 1 above)
Comment (TheWizardTower) [reply core-data/lib/Core/Data/Structures.hs:202]: I thought I'd removed that change, it doesn't compile (it considers `Key` redundant, but then wants a bunch of other things, like `Ord` and `Hashable` and others, so I gave up chasing all of them).

**Comment by TheWizardTower** [comment]
Mischief Managed, in https://github.com/aesiniath/unbeliever/pull/97. I think this PR is ready to get monched into main.

**Pull request merged**